pyiron_atomistics/calphy/job.py

```diff
@@ -651,34 +721,82 @@ def collect_general_output(self):
             None
         """
         if self._data is not None:
+            #solid liquid specific outputs
             if "spring_constant" in self._data["average"].keys():
                 self.output.spring_constant = self._data["average"]["spring_constant"]
+            if "density" in self._data["average"].keys():
+                self.output.equilibrium_density = self._data["average"]["density"]
+            self.output.equilibrium_volume_atom = self._data["average"]["vol_atom"]
+
+
+            #main results from mode fe
+            self.output.temperature = self.input.temperature
+            self.output.pressure = self.input.pressure
             self.output.energy_free = self._data["results"]["free_energy"]
             self.output.energy_free_error = self._data["results"]["error"]
             self.output.energy_free_reference = self._data["results"][
                 "reference_system"
             ]
             self.output.energy_work = self._data["results"]["work"]
-            self.output.temperature = self.input.temperature
+            self.output.energy_pressure = self._data["results"]["pv"]
+
+            #collect ediffs and so on
             f_ediff, b_ediff, flambda, blambda = self.collect_ediff()
             self.output.forward_energy_diff = list(f_ediff)
             self.output.backward_energy_diff = list(b_ediff)
             self.output.forward_lambda = list(flambda)
             self.output.backward_lambda = list(blambda)
+
+            #get final structure
+            traj = Trajectory(os.path.join(self.working_directory, "conf.equilibration.dump"))
+            aseobj = traj[0].to_ase(species=self.calc.element)[0]
+            pyiron_atoms = ase_to_pyiron(aseobj)
+            self.output.structure_final = pyiron_atoms
+
             if self.input.mode == "ts":
```

Comment:
Maybe separate methods depending on `input.mode` would be more readable, but your call.

Reply:
I'm not sure I follow, the first part is general for all modes, then the specific parts depending on the mode.

Reply:
Yeah, I meant to put the parts in the if clauses into separate methods, but as I said no strong opinions.

Reply:
Thanks!